projects/plugins/jetpack/tests/e2e/specs/connection.test.js

```diff
@@ -66,4 +76,45 @@ describe( 'Connection', () => {
 			expect( await jetpackPage.isConnected() ).toBeTruthy();
 		} );
 	} );
+
+	it( 'In-place upgrading a plan from Security Daily to Complete', async () => {
+		await step( 'Can set a sandbox cookie', async () => {
+			const sidebar = await Sidebar.init( page );
+			await sidebar.setSandboxModeForPayments( cookie );
+			await sidebar.setSandboxModeForPayments( cookie, '.cloud.jetpack.com' );
+			await sidebar.setSandboxModeForPayments( cookie, '.' + new URL( siteUrl ).host );
+		} );
+
+		await step( 'Can start in-place connection', async () => {
+			await ( await Sidebar.init( page ) ).selectJetpack();
+			await doInPlaceConnection( 'security' );
+		} );
+
+		await step( 'Can process payment for Security Daily plan', async () => {
+			await ( await CheckoutPage.init( page ) ).processPurchase( cardCredentials );
+			await ( await ThankYouPage.init( page ) ).waitForSetupAndProceed();
+			await ( await MyPlanPage.init( page ) ).returnToWPAdmin();
+		} );
+
+		await step( 'Can assert that site has a Security Daily plan', async () => {
+			const jetpackPage = await JetpackPage.init( page );
+			expect( await jetpackPage.isPlan( 'security' ) ).toBeTruthy();
+		} );
+
+		await step( 'Can visit plans page and select a Complete plan', async () => {
```

Comment:
yeah, ideally I'd agree.

> Test 2: having a plan different than complete, user can upgrade to complete

to have a "plan different than complete" it needs to be purchased, basically repeating all the steps before this one.

Reply:
@adimoldovan can you elaborate?

Reply:
I was thinking that maybe the upgrade test can be a "post-connection" test and rely on `maybePreConnect` method. It will upgrade from the existing plan (checking != 'complete').